jax/_src/nn/functions.py

```diff
@@ -912,19 +912,25 @@ def dot_product_attention(
   Returns:
     An array of the attention output with the same shape as :code:`query`.
   """
+  original_shape = jnp.asarray(query).shape
+  def _preprocess_array(t):
+    if t is None:
```

Comment:
Nit: I would personally move this out, since only `bias` and `mask` can be None.

Reply:
Done.